### PR TITLE
Fix transpiler_native

### PR DESCRIPTION
**Pull request by Simone-Bordoni**
-Fix CU3 and CU2 native gate decomposition
-Added T, Tdg, S, Sdg and TOFFOLI native gate decomposition
-Added tests
-Modified tests, now native decomposition and original gates are compared excluding global phase factor

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [X] Tests are passing.
- [X] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Review by stavros11 — APPROVED** [review]
Looks good to me, thanks!

File: src/qibolab/tests/test_transpilers_native.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from qibo import gates
 from qibo.backends import NumpyBackend
 from qibo.models import Circuit
+from scipy.linalg import expm
```
[the remark this answers is not in the document]
Comment (Simone-Bordoni) [reply src/qibolab/tests/test_transpilers_native.py:6]: it is not used, tomorrow i can remove it

[the remark this answers is not in the document]
Comment (Simone-Bordoni) [reply src/qibolab/tests/test_transpilers_native.py:6]: I can also remove the "phase" parameter in "assert_matrices_allclose" cause now it is not used anymore

[the remark this answers is not in the document]
Comment (stavros11) [reply src/qibolab/tests/test_transpilers_native.py]: Great, thanks!

**Pull request merged**